python/plugins/processing/algs/qgis/SplitWithLines.py

```diff
@@ -81,107 +81,111 @@ def processAlgorithm(self, progress):
         else:
             total = 100.0 / float(len(features))
 
-        multiGeoms = 0 # how many multi geometries were encountered
-
         for current, inFeatA in enumerate(features):
             inGeom = inFeatA.geometry()
+            attrsA = inFeatA.attributes()
+            outFeat.setAttributes(attrsA)
 
             if inGeom.isMultipart():
-                multiGeoms += 1
-                # MultiGeometries are not allowed because the result of a splitted part cannot be clearly defined:
-                # 1) add both new parts as new features
-                # 2) store one part as a new feature and the other one as part of the multi geometry
-                # 2a) which part should be which, seems arbitrary
+                inGeoms = []
+
+                for g in inGeom.asGeometryCollection():
+                    inGeoms.append(g)
             else:
-                attrsA = inFeatA.attributes()
-                outFeat.setAttributes(attrsA)
                 inGeoms = [inGeom]
-                lines = spatialIndex.intersects(inGeom.boundingBox())
 
-                if len(lines) > 0:  # has intersection of bounding boxes
-                    splittingLines = []
+            lines = spatialIndex.intersects(inGeom.boundingBox())
+
+            if len(lines) > 0:  # has intersection of bounding boxes
+                splittingLines = []
+
+                engine = QgsGeometry.createGeometryEngine(inGeom.geometry())
+                engine.prepareGeometry()
 
-                    engine = QgsGeometry.createGeometryEngine(inGeom.geometry())
-                    engine.prepareGeometry()
+                for i in lines:
+                    try:
+                        splitGeom = splitGeoms[i]
+                    except:
+                        continue
 
-                    for i in lines:
-                        try:
-                            splitGeom = splitGeoms[i]
-                        except:
+                    # check if trying to self-intersect
+                    if sameLayer:
+                        if inFeatA.id() == i:
                             continue
 
-                        # check if trying to self-intersect
-                        if sameLayer:
-                            if inFeatA.id() == i:
+                    if engine.intersects(splitGeom.geometry()):
+                        splittingLines.append(splitGeom)
+
+                if len(splittingLines) > 0:
+                    for splitGeom in splittingLines:
+                        splitterPList = None
+                        outGeoms = []
+
+                        split_geom_engine = QgsGeometry.createGeometryEngine(splitGeom.geometry())
+                        split_geom_engine.prepareGeometry()
+
+                        while len(inGeoms) > 0:
+                            inGeom = inGeoms.pop()
+
+                            if inGeom.geometry() == None: # this has been encountered and created a run-time error
```

Comment:
Did you do inGeom.geometry().isEmpty() or inGeom.isEmpty() (should be the second)